### ISSUE TITLE
Terraform 0.14 -  Azure API Management resource fails

### ISSUE THREAD

**Issue by prabhakarreddy1234** [terraform-providers/terraform-provider-azurerm#9807]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_api_management`

### Terraform Configuration Files

```
resource "azurerm_api_management" "platform" {
  name                = local.gateway_name
  resource_group_name = azurerm_resource_group.gateway-persistent.name
  location            = var.gateway_region
  publisher_name      = var.gateway_publisher_name
  publisher_email     = var.gateway_publisher_email

  sku_name = var.gateway_sku_name

  hostname_configuration {
    developer_portal {
      host_name           = var.gateway_developer_portal_host
    }
    proxy {
      host_name            = var.gateway_proxy_host_name
      certificate          = filebase64("${var.tls_certificate_portal_wildcard_file_path}")
      certificate_password = var.tls_certificate_portal_wildcard_password
    }
  }

  protocols {
    enable_http2 = true
  }

  tags = local.common_tags

  lifecycle {
    ignore_changes = [
      tags
    ]
  }
}
```

### Debug Output

```
Error: expected "hostname_configuration.0.proxy.0.key_vault_id" to not be an empty string, got
  on api-management.tf line 1, in resource "azurerm_api_management" "platform":
   1: resource "azurerm_api_management" "platform" {
```
![image](https://user-images.githubusercontent.com/1984707/101791261-3b3d2f80-3afb-11eb-87c3-291df0bd74cb.png)

### Expected Behaviour

It should create API Management resource like it does with terraform 0.13.5

### Actual Behaviour

It throws the error shown above with Terraform 0.14.0 but works fine with 0.13.5

### Steps to Reproduce

1. `terraform apply`

**Comment by tombuildsstuff** [comment]
hi @prabhakarreddy1234 

Thanks for opening this issue.

Taking a look through here I believe this has been fixed in version 0.14.2 of Terraform Core - would you be able to upgrade and see if that works for you?

Thanks!

**Comment by prabhakarreddy1234** [comment]
@tombuildsstuff  Thanks.  It works well with 0.14.2 .
